services/image-rec-auto/test/test_service.py

```diff
@@ -0,0 +1,277 @@
+"""Basic tests to ensure service functionality."""
+
+__author__ = "Kevin Li and Alex Witt"
+
+import copy
+import io
+import service
+from typing import Optional
+import unittest
+from unittest import mock
+
+from hawk_eye.inference import types
+from PIL import Image
+
+from messages import imagery_pb2
+from messages import image_rec_pb2
+from . import test_util
+
+
+class TestBase(unittest.TestCase):
+    """Form a class which intializes the shared items across the service tests."""
+
+    imagery_host = "imagery"
+    imagery_port = 1234
+    master_host = "image-rec-master"
+    master_port = 1234
+    image_id = 0
+    url = f"http://{imagery_host}:{imagery_port}/api/image/{image_id}"
+
+    auto_service = service.Service(
+        imagery_host=imagery_host,
+        imagery_port=imagery_port,
+        master_host=master_host,
+        master_port=master_port,
+        fetch_interval=1,
+    )
+    image_msg = imagery_pb2.Image()
+    image_msg.id = image_id
+    byteIO = io.BytesIO()
+    field_image = Image.open(test_util.TARGET_FIXTURE)
+    field_image.save(byteIO, format="JPEG")
+    image_msg.image = byteIO.getvalue()
+
+    def _mock_response(
+        self,
+        status: Optional[int] = 200,
+        content: Optional[str] = None,
+        headers: Optional[dict] = None,
+        ok: Optional[bool] = True
+    ) -> mock.Mock:
+        """Generalize function to mock the get/post calls in the service."""
+
+        mock_resp = mock.Mock()
+        # Set status code and content.
+        mock_resp.status_code = status
+        mock_resp.content = content
+
+        if headers:
+            mock_resp.headers = copy.deepcopy(headers)
+
+        return mock_resp
+
+
+class TestImageRec(TestBase):
+
+    @mock.patch.object(service.Service, "_get_next_id")
+    @mock.patch.object(service.Service, "_get_image")
+    @mock.patch.object(service.Service, "_queue_targets")
+    @mock.patch.object(service.Service, "_finish_processing")
+    def test_target_rec(
+        self,
+        mock_finish_processing: mock.MagicMock,
+        mock_queue_targets: mock.MagicMock,
+        mock_get_image: mock.MagicMock,
+        mock_get_next_id: mock.MagicMock,
+    ) -> None:
+
+        mock_get_next_id.return_value = 0
+        mock_get_image.return_value = self.image_msg
+
+        def _queue_side_effect(image_id, _image_proto, _image, targets):
+            self.assertEqual(image_id, 0)
+            self.assertEqual(len(targets), 1)
+
+            target = targets[0]
+            self.assertNotEqual(target, None)
+            self.assertEqual(target.shape, types.Shape.TRAPEZOID)
+            self.assertEqual(target.background_color, types.Color.NONE)
```

Comment:
For lines 88-90, should we leave these lines out in case a future release detects the alphanumeric/color?

Reply:
I say leave it for now. It'll probably be a while before hawk_eye can accurately return those attributes, and at that point, it'll be worth reviewing this code to make sure everything work right.